### ISSUE TITLE
Bug: Content from 404 override controller gets discarded

### ISSUE THREAD

**Issue by PurpleVsGreen** [codeigniter4/CodeIgniter4#5869]
### PHP Version

7.4

### CodeIgniter4 Version

4.1.9

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

apache

### Database

none

### What happened?

No output on 404 error

### Steps to Reproduce

0. Place this in `app/Config/Routes.php`:
```php
$routes->set404Override('App\\Controllers\\NotFound');
```
1. Create the NotFound controller, in `app/Controllers/NotFound.php`:
```php
<?php

namespace App\Controllers;

class NotFound extends BaseController
{
    public function index()
    {
        $this->response->setStatusCode(404)->setBody(
            view('page_not_found')
        );
    }
}
```
2. Create the `app/Views/page_not_found.php` file with some content

### Expected Output

Contents of `page_not_found.php`

### Anything else?

Any returned string or ResponseInterface instance from the controller is discarded as well. When the route matches a controller, any returned string is used as the response body. Returning the response object would work fine as well.

**Comment by PurpleVsGreen** [comment]
Commit 44a35f5 fixes this

**Comment by kenjis** [comment]
It was fixed in `develop` as you pointed out.
Duplicate of #5697.

**Comment by PurpleVsGreen** [comment]
Thank you so much! I'm really glad that the issue has been resolved and am looking forward to the next release.

Interestingly, the commit that solved it was nearly bit identical to the way I solved it after filing this bug report. So to my surprise, I actually thought I was looking at the wrong source file after forking the repository to submit a PR. Kudos to the team, especially since a regression test was added to ensure that the bug doesn't reappear.
